compiler-utils/src/testFixtures/java/com/sqareup/anvil/compiler/internal/testing/TestUtils.kt

```diff
@@ -0,0 +1,181 @@
+@file:Suppress("unused")
+
+package com.sqareup.anvil.compiler.internal.testing
+
+import com.google.common.truth.Truth.assertThat
+import com.squareup.anvil.annotations.ExperimentalAnvilApi
+import com.squareup.anvil.compiler.AnvilCommandLineProcessor
+import com.squareup.anvil.compiler.AnvilComponentRegistrar
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.KotlinCompilation.Result
+import com.tschuchort.compiletesting.PluginOption
+import com.tschuchort.compiletesting.SourceFile
+import dagger.Component
+import dagger.Module
+import dagger.Subcomponent
+import dagger.internal.codegen.ComponentProcessor
+import org.jetbrains.kotlin.config.JvmTarget
+import java.io.File
+import java.io.OutputStream
+import java.util.Locale
+import kotlin.reflect.KClass
+
+/**
+ * Helpful for testing code generators in unit tests end to end.
+ */
+@ExperimentalAnvilApi
+public fun compileAnvil(
+  vararg sources: String,
+  enableDaggerAnnotationProcessor: Boolean = false,
+  generateDaggerFactories: Boolean = false,
+  generateDaggerFactoriesOnly: Boolean = false,
+  disableComponentMerging: Boolean = false,
+  allWarningsAsErrors: Boolean = true,
+  useIR: Boolean = true,
+  messageOutputStream: OutputStream = System.out,
+  block: Result.() -> Unit = { }
+): Result {
+  return KotlinCompilation()
+    .apply {
+      compilerPlugins = listOf(AnvilComponentRegistrar())
+      this.useIR = useIR
+      useOldBackend = !useIR
```

Comment:
Is there a backward compatibility reason for keeping this around? No way of just using the new IR by default?

Reply:
We're still using the old backend in our codebase I'm sure others do as well. I want to keep compatibility for a little longer until I rip out the code.